internal/controllers/webhook_controller.go

```diff
@@ -50,32 +54,70 @@ type CelRequestPayload struct {
 
 // RegisterWebhook godoc
 // @Summary      Register a new webhook
-// @Description  Registers a new webhook with the specified configuration.
+// @Description  Registers a new webhook with the specified configuration. The target URI is validated to ensure it is a valid URL, responds with 200 within a timeout, and returns a verification token.
 // @Tags         Webhooks
 // @Accept       json
 // @Produce      json
 // @Param        request  body      object  true  "Request payload"
 // @Success      201      "Webhook registered successfully"
-// @Failure      400      "Invalid request payload"
+// @Failure      400      "Invalid request payload or target URI"
 // @Failure      500      "Internal server error"
 // @Security     BearerAuth
 // @Router       /webhooks [post]
 func (w *WebhookController) RegisterWebhook(c *fiber.Ctx) error {
 	type RequestPayload struct {
-		Service     string `json:"service" validate:"required"`
-		Data        string `json:"data" validate:"required"`
-		Trigger     string `json:"trigger" validate:"required"`
-		Setup       string `json:"setup" validate:"required"`
-		Description string `json:"description"`
-		TargetURI   string `json:"target_uri" validate:"required"`
-		Status      string `json:"status"`
+		Service           string `json:"service" validate:"required"`
+		Data              string `json:"data" validate:"required"`
+		Trigger           string `json:"trigger" validate:"required"`
+		Setup             string `json:"setup" validate:"required"`
+		Description       string `json:"description"`
+		TargetURI         string `json:"target_uri" validate:"required"`
+		Status            string `json:"status"`
+		VerificationToken string `json:"verification_token" validate:"required"`
 	}
 	var payload RequestPayload
 	if err := c.BodyParser(&payload); err != nil {
 		w.logger.Error().Err(err).Msg("Invalid request payload")
 		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid request payload"})
 	}
 
+	// --- Begin URI Validation ---
+
+	// 1. Validate the target URI is a well-formed URL
+	parsedURL, err := url.ParseRequestURI(payload.TargetURI)
+	if err != nil {
+		w.logger.Error().Err(err).Msg("Invalid target URI")
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"error": "Invalid target URI"})
+	}
+
+	// 2. Test the target URI with an HTTP GET request with a timeout
+	client := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+	resp, err := client.Get(parsedURL.String())
```

Comment:
This is a GET request but typically webhooks are POST requests, unless if we specifically instruct developers they must have a GET request equivalent for verification... But I'm not sure I'd recommend that from a devex point of view. I think these should be changed to POST, and consider re-using part of the logic for posting to webhook URI's. I would post the same schema we'd normally do but with dummy data.